### DISCUSSION TITLE
Copilot Extension Error: Server or Network Issue?

### DISCUSSION THREAD

**Discussion by jiemoL** [community/community#116532] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

Hey, everyone! I'm encountering a puzzling issue with the GitHub Copilot extension in VS Code, and I'm not sure if it's a problem with the server or my network. I'd appreciate it if anyone could offer some insights or suggestions.

Here's the error log I'm getting:

2024-04-02 20:52:05.009 [error] [ghostText] Error on ghost text request: FetchError: connect ETIMEDOUT 20.221.80.166:443  
at fetch (c:\Users\a2425\.vscode\extensions\github.copilot-1.177.0\node_modules\@adobe\helix-fetch\src\fetch\index.js:99:11)  
...  
{  
  type: 'system',  
  _name: 'FetchError',  
  code: 'ETIMEDOUT',  
  errno: -4039,  
  erroredSysCall: 'connect'  
}
This seems to indicate that Copilot is experiencing a timeout (ETIMEDOUT) when trying to connect to server 20.221.80.166:443. This usually indicates a network connection problem, but I checked my network connection and nothing seems to be wrong.

I would be grateful for any suggestions. Thanks in advance for your help!

**Comment by TaoYuLiann** [comment]
![image](https://github.com/community/community/assets/162962934/2d8bd15f-ea5a-45b8-beff-772b9516b2d5)
https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
The browser accesses this address and copies the remote address into VScode's settings [copilot http proxy], which works for me

**Reply by jiemoL** [reply]
My Rhost is 127.0.0.1:7890

**Reply by TaoYuLiann** [reply, marked as answer]
then set http://127.0.0.1:7890 to vscode settings [http proxy]

**Reply by jiemoL** [reply]
I reconfigured it and now Copilot works!

**Reply by jiemoL** [reply]
thank you
